app/controllers/forums/topics_controller.rb

```diff
@@ -164,19 +164,16 @@ def unsubscribe
 
 private
   def load_forum
-    @forum = ForumForum.find_using_slug(params[:forum_id])
-    raise ActiveRecord::RecordNotFound unless @forum
+    @forum = @course.forums.find_using_slug!(params[:forum_id])
   end
 
   def load_topic
-    @topic = ForumTopic.find_using_slug(params[:id] || params[:topic_id])
```

Comment:
This would however break creating new topics. This needs to flow through to the following if block